### DISCUSSION TITLE
ESP32 S3 compile and boot errors

### DISCUSSION THREAD

**Discussion by a-sapuppo** [micropython/micropython#11884] [ESP32]
i have a ESP32-S3-WROOM-1U-N4R8 module: ESP32-S3 4Gb flash 8Gb SPIRAM

i compile with docker and flash with no errors, boot messages loop with:
```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0xa (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0xf58
load:0x403c9700,len:0xc3c
load:0x6c696166,len:0x77206465
Invalid image block, can't boot.
ets_main.c 329 
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0xa (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0xf58
load:0x403c9700,len:0xc3c
load:0x6c696166,len:0x77206465
Invalid image block, can't boot.
ets_main.c 329 
```

**Comment by puppet13th** [comment]
You should flash `firmware.bin` from build folder using address `0x0`

**Comment by a-sapuppo** [comment]
bootloader from 0x0 ? not  0x1000.
```

python3 -m esptool --port /dev/ttyACM0 -b 460800 --before default_reset \
  --after hard_reset --chip esp32s3  write_flash --flash_mode dio --flash_size detect \
  --flash_freq 40m 0x1000 build-ESP32S3/bootloader/bootloader.bin \
  0x8000 build-ESP32S3/partition_table/partition-table.bin \
  0x10000 build-ESP32S3/micropython.bin
```

**Comment by puppet13th** [comment]
flash command for esp32:
`Using command -p COM72 --baud 921600 write_flash 0x1000 d:/mpy/firmware/GENERIC-1.20.0-4.4.4.bin`

flash command for esp32-s3:
`Using command -p COM82 --baud 921600 write_flash 0x0 d:/mpy/firmware/LOLIN_S3-1.20.0-4.4.4.bin` 

fresh boot:
```
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0xf34
load:0x403c9700,len:0xc3c
load:0x403cc700,len:0x2d98
entry 0x403c98f0
MicroPython v1.20.0 on 2023-05-01; LOLIN S3 with ESP32S3
Type "help()" for more information.
```

**Comment by a-sapuppo** [comment]
i cannot use registered boards. i have new board now: https://namino.cc/boards/namino-rosso/ and compile / flash from source. I have console on UART 1 pins 17/18 on this new board

**Comment by a-sapuppo** [comment]
do you want be a beta tester ? https://forum.namino.cc/categories

**Comment by puppet13th** [comment]
your board specs same as s3 board except the flash size. you can use 'GENERIC_S3_SPIRAM' board with some adjustment.

copy 'board/GENERIC_S3_SPIRAM' to 'board/ESP32S3'

edit 'board/ESP32S3/sdkconfig.board':
```
CONFIG_FLASHMODE_QIO=y
CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
CONFIG_ESPTOOLPY_FLASHSIZE_DETECT=y
CONFIG_ESPTOOLPY_AFTER_NORESET=y

CONFIG_SPIRAM_MEMTEST=

CONFIG_ESPTOOLPY_FLASHSIZE_4MB=y
CONFIG_ESPTOOLPY_FLASHSIZE_8MB=
CONFIG_ESPTOOLPY_FLASHSIZE_16MB=
CONFIG_PARTITION_TABLE_CUSTOM=y
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions.csv"
```

then flash `firmware.bin` using `0x0` address.

for console output, did you use external uart or directly from the usb( uart from the board )
